### PR TITLE
fix(training): handle packed SFT CP masks

### PR DESCRIPTION
**Pull request by cuichenx**
## Summary

- Makes packed SFT batches return `attention_mask=None` instead of a 1-D sentinel tensor, matching the packed THD path where no conventional attention mask is used.
- Derives `dataset.packed_sequence_specs.pad_seq_to_mult` in the Nemotron Ultra SFT launcher for CP runs; for `TP=2 CP=2 SP=True`, this resolves to `4`.
- Adds focused regression coverage for the packed attention-mask behavior and CP partition path.

## Root Cause

During Lyris CP2 validation for the r0.5.0 Ultra backport, the run failed before iteration 1 in `_partition_packed_batch_for_cp()` at `val.size(1)`. Packed SFT collate was emitting `attention_mask=torch.LongTensor([1])`; CP2 middle PP stages keep full packed metadata and attempted to partition that 1-D tensor along dimension 1.

## Validation

- `git diff --check origin/main...HEAD`
- `bash -n examples/models/nemotron/nemotron_3/ultra/slurm_sft.sh`
- `python3 -m py_compile src/megatron/bridge/data/datasets/sft.py tests/unit_tests/data/datasets/test_chat_template.py tests/unit_tests/training/test_gpt_step.py`
- `pre-commit run --all-files`
- Attempted `uv run pre-commit run --all-files`, but local host dependency setup failed while building TransformerEngine because its isolated build env did not include `setuptools`.
- Lyris GB200 QA on the r0.5.0 stack: replacement CP2 job `2212819` generated/resolved `pad_seq_to_mult4` OpenMath packed files and reached iteration `64/1000` with `0` skipped and `0` NaN iterations; the previous `IndexError` did not recur.
- Lyris GB200 CP1 comparison job `2212274` remained healthy at iteration `861/1000`, also with `0` skipped and `0` NaN iterations.

### CODE REVIEW

**Review by yaoyu-33 — APPROVED** [review]

**Comment by suiyoubi** [comment]
/ok to test [f699f3d](https://github.com/NVIDIA-NeMo/Megatron-Bridge/pull/4519/commits/f699f3df6ebb23c7dccd289e75520c959f403d06)

**Comment by suiyoubi** [comment]
faling CI should be resolved by https://github.com/NVIDIA-NeMo/Megatron-Bridge/pull/4516. Re-triggering the tests
